retail/interactive-tutorial/RetailProducts.Samples/CrudProductSample.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_crud_product]
+// Create product in a catalog using Retail API
+
+using Google.Cloud.Retail.V2;
+using System;
+
+/// <summary>
+/// The crud product sample class.
+/// </summary>
+public class CrudProductSample
+{
+    /// <summary>
+    /// Generate the product.
+    /// </summary>
+    /// <returns>Generated product.</returns>
+    private static Product GenerateProduct()
+    {
+        return new Product
+        {
+            Id = Guid.NewGuid().ToString("N").Substring(0, 14),
+            Title = "Nest Mini",
+            Type = Product.Types.Type.Primary,
+            PriceInfo = new PriceInfo
+            {
+                Price = 30.0f,
+                OriginalPrice = 35.5f,
+                CurrencyCode = "USD"
+            },
+            Availability = Product.Types.Availability.InStock,
+            Categories = { "Speakers and displays" },
+            Brands = { "Google" }
+        };
+    }
+
+    /// <summary>
+    /// Get the product for update.
+    /// </summary>
+    /// <param name="productName">The actual name of the retail product.</param>
+    /// <returns>Generated product for update.</returns>
+    private static Product GenerateProductForUpdate(string productName)
+    {
+        return new Product
+        {
+            Id = Guid.NewGuid().ToString("N").Substring(0, 14),
```

Comment:
Why do you have to add an ID here? Shouldn't you just use product name? Else, wouldn't you just create a new product (given that you have set allow missing to true)
```suggestion
```